pkg/api/v20220904/clustermanager.go

```diff
@@ -0,0 +1,192 @@
+package v20220904
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// OCM Kinds supported
+const (
+	MachinePoolKind          = "MachinePool"
+	SyncIdentityProviderKind = "SyncIdentityProvider"
+	SyncSetKind              = "SyncSet"
+	SecretKind               = "Secret"
+)
+
+type ClusterManagerConfigurationList struct {
+	ClusterManagerConfigurations []*ClusterManagerConfiguration `json:"value"`
+
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+type ClusterManagerConfiguration struct {
+	// ID is the unique identifier for the cluster manager configuration
+	ID string `json:"id,omitempty"`
+
+	// The resource name
+	Name string `json:"name,omitempty"`
+
+	// The parent cluster resourceId
+	ClusterResourceId string `json:"clusterResourceId,omitempty"`
+
+	// If this object is marked to be deleted
+	Deleting bool `json:"deleting,omitempty"` // https://docs.microsoft.com/en-us/azure/cosmos-db/change-feed-design-patterns#deletes
+
+	// The ClusterManagerConfigurations properties
+	Properties ClusterManagerConfigurationProperties `json:"properties,omitempty"`
+
+	// SystemData metadata from ARM
+	SystemData *SystemData `json:"systemData,omitempty"`
+}
+
+type ClusterManagerConfigurationProperties struct {
+	Resources interface{} `json:"resources,omitempty"`
+}
+
+// SyncSetList represents a list of SyncSets
+type SyncSetList struct {
+	// The list of syncsets.
+	SyncSets []*SyncSet `json:"value"`
+
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// SyncSet represents a SyncSet for an Azure Red Hat OpenShift Cluster.
+type SyncSet struct {
+	// This is a flag used during the swagger generation typewalker to
+	// signal that it should be marked as a proxy resource and
+	// not a tracked ARM resource.
+	proxyResource bool
```

Comment:
not a fan of defining swagger specific fields within production-shipped code, but can address later.

Reply:
yeah its not great